pkg/controller/opentelemetrycollector/reconcile_service.go

```diff
@@ -2,19 +2,44 @@ package opentelemetrycollector
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	errs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/util/intstr"
 
 	"github.com/open-telemetry/opentelemetry-operator/pkg/apis/opentelemetry"
 	"github.com/open-telemetry/opentelemetry-operator/pkg/apis/opentelemetry/v1alpha1"
 )
 
+var (
+	// ErrInvalidYAML represents an error in the format of the configuration file
+	ErrInvalidYAML = errors.New("couldn't parse the opentelemetry-collector configuration to assemble a list of ports for the service")
+
+	// ErrNoReceivers indicates that there are no receivers in the configuration
+	ErrNoReceivers = errors.New("no receivers available as part of the configuration")
+
+	// ErrReceiversNotAMap indicates that the receivers property isn't a map of values
+	ErrReceiversNotAMap = errors.New("receivers property in the configuration doesn't contain valid receivers")
```

Comment:
nit: Would it be better to actually say same as comment, i.e. "... configuration isn't a map of values"?

Reply:
For someone reading the log messages, "map of values" might not make sense.